### DISCUSSION TITLE
question about alarm duration

### DISCUSSION THREAD

**Discussion by 1998729** [jertel/elastalert2#1035] [Q&A] [answered]
Excuse me, I think when a certain alarm lasts for how long to start the alarm, which word length should be used as the standard?

**Comment by 1998729** [comment]
For example: 1alert 3m query for 1m send alert
An alarm is queried once every three minutes. If it lasts for one minute, the alarm will be sent

**Reply by 1998729** [reply]
> For example: 1alert 3m query for 1m send alert An alarm is queried once every three minutes. If it lasts for one minute, the alarm will be sent

Can the minute here be defined in elastalert2? @jertel

**Comment by jertel** [comment, marked as answer]
If an alarm situation occurs due to multiple triggering events, you can use a frequency (or other rule types) to only trigger the alarm if the X number of triggering events have occurred. 

For example, suppose you have an event logged when the CPU exceeds 90%, and at most 1 event can be logged every 10 seconds. You could use a frequency rule to check for a minimum of 6 "high CPU usage" events in a 1 minute timeframe. If only 3 events are logged in that 1m timeframe then no alarm will be sent. But if 6 events are logged in that 1m timeframe than the alarm will be sent.

However, this might not be possible if your source/triggering events are solitary events. One option might be to write a rule filter that only triggers the alarm if the triggering event is seen within a 1m window, but the clearing event is not seen in that same window:

```
query_string: _exists:"HIGH CPU USAGE" AND NOT _exists:"NORMAL CPU USAGE"
```

But this can be complicated if multiple triggering events occur consecutively, and then the order of the trigger event and the clear event isn't available.

**Reply by 1998729** [reply]
> If an alarm situation occurs due to multiple triggering events, you can use a frequency (or other rule types) to only trigger the alarm if the X number of triggering events have occurred.
> 
> For example, suppose you have an event logged when the CPU exceeds 90%, and at most 1 event can be logged every 10 seconds. You could use a frequency rule to check for a minimum of 6 "high CPU usage" events in a 1 minute timeframe. If only 3 events are logged in that 1m timeframe then no alarm will be sent. But if 6 events are logged in that 1m timeframe than the alarm will be sent.
> 
> However, this might not be possible if your source/triggering events are solitary events. One option might be to write a rule filter that only triggers the alarm if the triggering event is seen within a 1m window, but the clearing event is not seen in that same window:
> 
> ```
> query_string: _exists:"HIGH CPU USAGE" AND NOT _exists:"NORMAL CPU USAGE"
> ```
> 
> But this can be complicated if multiple triggering events occur consecutively, and then the order of the trigger event and the clear event isn't available.

For example, if a certain threshold lasts for 5 minutes, an alarm will be issued.
